yroots/utils.py

```diff
@@ -1209,3 +1209,99 @@ def Df(x):
+    
+    A tolerance may be a float or an iterable type (ex. list, numpy array, etc). If an iterable
+    is used, all iterables passed in must be the same length. Any floats passed in will be
+    resized into lists of the same length as the iterables being used.
+    
+    If a tolerance of name "tol" is passed in, the object self.tols is created to store all
+    or the values to use for "tol". self.tol will be the current value for "tol".
+
+    ---DEVELOPER WARNING---
+    IF ANY OTHER ATRRIBUTE IS CREATED OF ITERABLE TYPE THIS CLASS MAY CRASH!!!
+
+    Attributes
+    ----------
+    numTols: int
+        The number of tolerances that are being used.
+    currTol : int
+        The current tolerances to check
+    **tols : list
+        The list of tolerances to be used.
+    *tol : int
+        The next tolerance to use.
+        
+    Methods
+    -------
+    __init__
+        Initializes everything.
+    nextTols
+        Sets up the next tolerances to be used
+    '''
+    def __init__(self, **tolerances):
+        numTols = 1
+        #Finds the number of the tolerances to be used.
+        for name in tolerances:
+            if not isNumber(tolerances[name]):
+                numTols = len(tolerances[name])
+        
+        for name in tolerances:
+            value = tolerances[name]
+            if isNumber(value): #Turns the number into a list of the right name. Stores as attribute.
+                self.__setattr__(name+'s', [value]*numTols)
+            elif hasattr(value, '__iter__'): #Makes sure the list is the right length. Stores as attribute.
+                self.__setattr__(name+'s', value)
+                if len(value) != numTols:
+                    raise ValueError("Length of tolerence lists must be the same!")
+            else:
+                raise TypeError("Tolerance value must be number type or iterable!")
+
+        self.currTol = -1
+        self.numTols = numTols
+    
+    def nextTols(self):
+        """Determines the next tolerances
+
+        Returns
+        -------
+        nextTols : bool
+            True if there are more tols to run, otherwise False.
+        """
+        self.currTol += 1
+        if self.currTol >= self.numTols: #Returns False if there are no more tols to be used
+            return False
+        else:
+            names = []
+            vals = []
+            for name in self.__dict__:
+                #Finds every iterable type being stored
+                if not hasattr(self.__dict__[name], '__iter__'):
+                    continue
+                #Finds the next tol and stores it
+                names.append(name[:-1])
+                vals.append(self.__dict__[name][self.currTol])
+            #The storing is done outside the loop so the dictionary size doesn't change during iteration.
+            for name, val in zip(names, vals):
+                self.__setattr__(name, val)                
+            return True
```

Comment:
Some of the way this is written might be a bit confusing. The idea is you pass in as many tolerance arguments as you want. ex (tol1=5, tol2=7, tol3=[1,2,3]). One of the tolerances is a list so it creates the members tol1s = [5,5,5], tol2s = [7,7,7], tol3s = [1,2,3]. Then nextTols updates the values of tol1, tol2, tol3 to iterate through those lists, returning False when it's done. Basically just an easy way to handle lot of tolerances and make passing them through the code easier and cleaner.